shared/utils/yaml-to-shorthand.py

```diff
@@ -323,8 +322,8 @@ def to_xml_element(self):
         title.text = self.title
         add_sub_element(group, 'description', self.description)
 
-        if self.id_ is not "conditional_clause":
-            group = add_warning(group, self.warning)
+        if self.id_ != "conditional_clause":
+            add_warnings(group, self.warnings)
```

Comment:
conditional_clause Group?!

Reply:
@mpreisler yes.... conditional_clause kept showing up and throwing errors.

Reply:
but its warnings are going to be empty list, right? the issue is elsewhere, this just hides it.

the issue does not originate in this PR, it looks like we do not init all membes in all the ctors. once you fix that it will work.

Reply:
as I said, the problem is that the constructors (__init__ methods) don't initialize all members. fix that and the issue will go away - no need for the self.id_ checks above.

Reply:
conditional_clause is there for OCIL, ocil won't work without it AFAIK